OsmAnd-java/src/main/java/net/osmand/router/HHRoutePlanner.java

```diff
@@ -896,6 +902,13 @@ private T runRoutingWithInitQueue(HHRoutingContext<T> hctx) throws SQLException,
 			if (progress != null && progress.isCancelled) {
 				return null;
 			}
+			if (progress != null && straightDistanceStartEnd > 0) {
+				// approximate correlation between the distance and vertices (enough for progress bar)
+				double k = (double) hctx.stats.visitedVertices * 11 / straightDistanceStartEnd;
+				if (k < 1.0) {
+					progress.hhIterationProgress(k);
```

Comment:
Strange if

Reply:
removed (k is actually validated inside called method)